tests/modules/commandfilter.c

```diff
@@ -89,6 +89,29 @@ int CommandFilter_LogCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int
     return REDISMODULE_OK;
 }
 
+static int brpoplpush_count = 0;
+
+/* Filter to protect against Bug #11894 reappearing */
+void CommandFilter_BlpopFilter(RedisModuleCommandFilterCtx *filter)
+{
+    if (RedisModule_CommandFilterArgsCount(filter) != 3)
+        return;
+
+    const RedisModuleString *arg = RedisModule_CommandFilterArgGet(filter, 0);
+    size_t arg_len;
+    const char *arg_str = RedisModule_StringPtrLen(arg, &arg_len);
+
+    if (arg_len != 10 || memcmp(arg_str, "brpoplpush", 10))
+        return;
+
+    brpoplpush_count++;
+
+    if (brpoplpush_count % 2 != 0)
```

Comment:
I am not sure about this methodology. This is for tests purpose so there is no real harm, but in case the same function will be used in cases of timeout brpoplpush the counter will be messed up. can you just do an RM_CALL to incr some variable and check it's value after the client is unblocked?

Reply:
so I agree that its not great for general purpose (heck, 2 blockers, the 2nd wouldn't necc execute to perhaps the 3rd pass through).  The thought was just making some easiy predictable for testing purposes.  anything else just seems to complicate it beyond what's needed for testing purposes.

Reply:
can make one that reverses the direction.
i.e. changes BLMOVE RIGHT LEFT to BLMOVE LEFT RIGHT (swap the two arguments).
then to wake up the command we push two elements to the list, and watch which one is moved.

Reply:
@oranagra , I'll try that.